tests/application_store_tests/conftest.py

```diff
@@ -27,8 +31,25 @@ def app():
     yield create_app()
 
 
+class _ApplyFlaskClient(FlaskClient):
+    def open(
+        self,
+        *args: Any,
+        buffered: bool = False,
+        follow_redirects: bool = False,
+        **kwargs: Any,
+    ) -> TestResponse:
+        if "headers" in kwargs:
+            kwargs["headers"].setdefault("Host", Config.API_HOST)
+        else:
+            kwargs.setdefault("headers", {"Host": Config.API_HOST})
+        return super().open(*args, buffered=buffered, follow_redirects=follow_redirects, **kwargs)
+
+
 @pytest.fixture(scope="function")
-def flask_test_client(app):
+def flask_test_client():
+    app = create_app()
```

Comment:
Moving this from being injected to re-instantiated every time didn't feel good but something about the test layout + order causes half of them to fail as it was configured.